storage/jellyfish_merkle/src/node_type/mod.rs

```diff
@@ -583,4 +646,53 @@ pub enum NodeDecodeError {
     /// The first byte of the input is not a known tag representing one of the variants.
     #[fail(display = "lead tag byte is unknown: {}", unknown_tag)]
     UnknownTag { unknown_tag: u8 },
+
+    /// No children found in internal node
+    #[fail(display = "No children found in internal node")]
+    NoChildren,
+
+    /// Extra leaf bits set
+    #[fail(
+        display = "Non-existent leaf bits set, existing: {}, leaves: {}",
+        existing, leaves
+    )]
+    ExtraLeaves { existing: u16, leaves: u16 },
+}
+
+/// Helper function to serialize version in a more efficient encoding.
+/// We use a super simple encoding - the high bit is set if more bytes follow.
+fn serialize_u64_varint(mut num: u64, binary: &mut Vec<u8>) {
+    for _ in 0..8 {
+        let low_bits = num as u8 & 0x7f;
+        num >>= 7;
+        let more = (num > 0) as u8;
+        binary.push(low_bits | more << 7);
+        if more == 0 {
```

Comment:
I might prefer

```
let more = num > 0;
binary.push(low_bits | (more as u8) << 7);
if !more {
```

Seems a tiny bit more symmetric given that `more` in `deserialize_u64_varint` is a boolean.